### DISCUSSION TITLE
config_from_object() with namespace should only save the relevant key/value pairs

### DISCUSSION THREAD

**Discussion by jedie** [celery/celery#7035] [Ideas]
Common for django projects is:

`celery_app.config_from_object('django.conf:settings', namespace='CELERY')`

The problem here is IMHO that the complete django settings will be consume and not only the settings starts with `CELERY_`
e.g.: The output of **celery inspect report** it totally long and confusing.

And if may be a security issues that something like **SECRET_KEY** is consumed. (However, the entry in the report is replaced by `'********'`.

What's about to filter by namespace in the loader?

**Comment by auvipy** [comment]
could you proceed with the suggested improvements as a pr?

**Comment by auvipy** [comment]
moving this to discussion for now
